inductiva/tasks/run_simulation.py

```diff
@@ -35,57 +35,67 @@ def run_simulation(
 
-    if not format_utils.getenv_bool("DISABLE_TASK_METADATA_LOGGING", False):
-        metadata = {
-            "api_method_name": api_method_name.split(".")[1],
-            "machine_group_id": resource_pool_id,
-            "storage_dir": str(storage_dir),
-            **kwargs,
-        }
-        if extra_metadata is not None:
-            metadata = {**metadata, **extra_metadata}
-
-        with _metadata_lock:
-            _save_metadata(metadata,
-                           mode="w",
-                           path=pathlib.Path(input_dir) /
-                           TASK_METADATA_FILENAME_UPLOAD)
-
     task_id = api_invoker(
         api_method_name,
         params,
         type_annotations,
-        resource_pool_id=resource_pool_id,
+        resource_pool=computational_resources,
         storage_path_prefix=storage_dir,
     )
 
     if computational_resources is not None:
-        logging.info("Task submitted to machine group %s.",
-                     computational_resources.name)
+        logging.info("Task %s submitted to the queue of the %s.", task_id,
+                     computational_resources)
     else:
-        logging.info("Task submitted to the default resource pool.")
+        logging.info(
+            "Task %s submitted to the default queue. It will be picked for"
+            "execution whenever a computational resource is available.",
+            task_id)
 
     task = tasks.Task(task_id)
     if not isinstance(task_id, str):
         raise RuntimeError(
             f"Expected result to be a string with task_id, got {type(task_id)}")
 
     if not format_utils.getenv_bool("DISABLE_TASK_METADATA_LOGGING", False):
+        machine_group_id = None
+        if computational_resources is not None:
+            machine_group_id = computational_resources.id
+
+        metadata = {
+            "api_method_name": api_method_name.split(".")[1],
+            "machine_group_id": machine_group_id,
+            "storage_dir": str(storage_dir),
+            **kwargs,
+        }
+        if extra_metadata is not None:
+            metadata = {**metadata, **extra_metadata}
+
+        with _metadata_lock:
+            task_metadata_file = _save_metadata(metadata,
+                                                mode="w",
+                                                path=pathlib.Path(input_dir) /
+                                                TASK_METADATA_FILENAME_UPLOAD)
+
         with _metadata_lock:
-            _save_metadata({
+            global_metadata_file = _save_metadata({
                 **{
                     "task_id": task_id,
                     "input_dir": str(input_dir)
                 },
                 **metadata
             })
+        logging.info(
+            "Task configuration metadata is saved in a file in "
+            "the local input directory %s and added to the general "
+            "tasks metadata file in %s.", task_metadata_file,
+            global_metadata_file)
```

Comment:
I think there is no need to give details about the uploaded metadata file. In fact, I think it should be added only to the uploaded archive and not to the local input directory because if we submit multiple times the same input dir without any change to run in different machines, we'll only save the last one, even though multiple tasks used the same input.

```suggestion
        logging.info(
            "Task configuration metadata save to %s in the current working directory.",
            global_metadata_file)
```

Reply:
Unless the subsequent submissions append to the file in the input dir 🤔 ?

Reply:
But that data is still saved in the `global_metadata_file` so maybe the first is irrelevant. I think, one of the uses is the fact that it will be uploaded to the bucket together with the input files. Hence, when downloading the data it comes with it and you can manage things. However, technically you already that data saved in the input dir as well.

So maybe, it is too much information... wdyt @AugustoPeres?